Dockerfile.ci-unified

```diff
@@ -0,0 +1,52 @@
+ARG REGISTRY
+
+#Stage 1: Build the SRE Portal Assets
+FROM ${REGISTRY}/ubi8/go-toolset:1.20.12-5 as portal-build
+USER root
+
+#Copying package files and installing dependancies
+COPY portal/v2/package*.json ./
+RUN npm ci
+
+#Copying the rest of the source and build
+COPY portal/v2/ ./
+RUN npm run lint
+RUN mkdir -p /build/portal/v2/dist  # Ensure the directory exists
+RUN npm run build || (echo "Build failed, listing current directory:" && ls -la)
+
```

Comment:
bug: when I run this locally `make ci-rp` does not execute this stage. https://docs.docker.com/build/building/multi-stage/ is worth a read here to understand why (ignore the `buildkit` stuff - podman doesn't support a majority of it) :) and reach out if you need additional help.

Reply:
Fix this issue in my latest commit